lib/norm/contract.ex

```diff
@@ -0,0 +1,228 @@
+defmodule Norm.Contract do
+  @moduledoc """
+  Design by Contract with Norm.
+
+  This module provides a `@contract` macro that can be used to define specs for arguments and the
+  return value of a given function and optionally additional pre- and post-conditions.
+
+  Calling `use Norm.Contract` automatically imports all functions from the `Norm` module.
```

Comment:
this should be fine in majority of cases,I'm actually gonna change this to only `require Norm; import Norm, only: [spec: 2]` to limit possible conflicts (if a user happens to have e.g. valid?/2 in the module already, this won't compile). I'd be ok with not importing at all too.

Reply:
I think that's fine. For me in the majority of cases I import Norm directly into the module.

Reply:
Do you want to keep it as it is on branch, or how it was before, that `use Norm.Contract` did `import Norm`.

I just realised that the way it is on branch can has unintended consequences:

```elixir
defmodule Foo do
  use Norm.Contract
  import Norm

  def gen() do
    gen(spec(is_integer()))
  end
end
```

works well, **but** swapping order:

```elixir
  import Norm
  use Norm.Contract
```

un-imports Norm.gen/1 and so the code above won't compile. Not sure what's the best way forward, I can think of a couple ways:

  1. go back to what we had before, use Norm.Contract does import Norm
  2. document that order matters
  3. in `Norm.Contract.__using__` check if any Norm functions were imported and if so, preserve them (we can do this with `__CALLER__.functions[Norm]`

Option 2. is meh, 3. is I guess ok but not sure if it could cause even more unintended consequences.